src/logging/log_dest_file.cpp

```diff
@@ -136,6 +139,10 @@ void file_log_disable() {
     log_info(FileSystem, "Stopped logging to file");
 
     std::scoped_lock guard(mutex);
+    while (!data->buffer.isEmpty()) {
```

Comment:
Code duplication with `file_log_write`. Please unify this in a dedicated `file_log_write_buffer`. The overflow checking should be also included (but not the ferror)